### PR TITLE
[HttpKernel] Fix HttpCache validation HTTP method

### PR DESCRIPTION
**Pull request by tgalopin**
| Q | A |
| --- | --- |
| Branch? | 2.7 |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #19582 |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

**Comment by dunglas** [comment]
👍

File: src/Symfony/Component/HttpKernel/HttpCache/HttpCache.php [hunk 1]
```diff
@@ -374,7 +374,11 @@ protected function validate(Request $request, Response $entry, $catch = false)
         $subRequest = clone $request;
 
         // send no head requests because we want content
-        $subRequest->setMethod('GET');
+        if ($request->isMethodSafe() && $request->getMethod() !== 'HEAD') {
+            $subRequest->setMethod($request->getMethod());
```
Comment (fabpot) [inline src/Symfony/Component/HttpKernel/HttpCache/HttpCache.php]: Is this needed. By default, the subrequest is going to have the same method as the main one.

Comment (tgalopin) [inline src/Symfony/Component/HttpKernel/HttpCache/HttpCache.php]: Indeed, I updated this.

File: src/Symfony/Component/HttpKernel/HttpCache/HttpCache.php [hunk 2]
```diff
@@ -374,7 +374,9 @@ protected function validate(Request $request, Response $entry, $catch = false)
         $subRequest = clone $request;
 
         // send no head requests because we want content
-        $subRequest->setMethod('GET');
+        if (!$request->isMethodSafe() || $request->getMethod() === 'HEAD') {
```
Comment (fabpot) [inline src/Symfony/Component/HttpKernel/HttpCache/HttpCache.php]: I'm pretty sure that `isMethodSafe()` here will always return true. If not, we would not lookup anything in the cache in the first place. So, this part of the condition can be removed.

Comment (tgalopin) [inline src/Symfony/Component/HttpKernel/HttpCache/HttpCache.php]: Fixed

**Comment by fabpot** [comment]
Thank you @tgalopin.

**Pull request merged**